### ISSUE TITLE
Show in text form what the chosen discard gave up against the best one

### ISSUE THREAD

**Issue by markafitzgerald1** [markafitzgerald1/cribbage-trainer#10]
Add in order to let user know how many points they're expected to have lost relative to optimal play with each discard.

---

## Scope decision 2026-09-09

Approved by Mark Fitzgerald; wording by Claude Code (Opus 5, high effort). This resolves the question left open in the 2026-05-04 comment ("I'm not sure if we want to communicate how the user could discard better in this manner") and supersedes the three shapes offered in the 2026-09-09 design proposal above.

**The spoon-feeding concern does not apply to this surface.** Two facts settle it. `ScoredPossibleKeepDiscards` renders only when `discardIsComplete(dealtCards)` (`src/ui-react/Trainer.tsx`), so the analysis table is already post-commit only — there was never a pre-decision reveal to protect against. And the table already defaults to Net descending with the user's own row highlighted (`isHighlighted={scoredKeepDiscard.keep.every((card) => card.kept)}`), so the best option and the chosen option are both already on screen with their values printed. This issue removes one subtraction from numbers already displayed adjacent to each other; it cannot give away an answer the table has already given.

**Scoped intent.** After a discard is committed, show what that specific discard gave up against the best available one — the number the app already computes and records — so the cost of a decision is visible on the hand it was made on, and not only as a running average and a chart point.

**Cut from scope: a delta on all fifteen rows.** Two reasons. It is the half that risks false precision, since most inter-row gaps are smaller than the estimates' own error — which is the exact problem #627, #725 and #774 exist to fix, so shipping fifteen two-decimal deltas now would work against the trust gate while appearing to serve it. And it is the half #11 actually needs: #11 wants this issue only for "a text equivalent, so the colour scale is not the only way", which is an accessibility requirement of the colour scale and belongs with the person weighing colour blindness. The fifteen-row form therefore moves to #11 rather than staying here, and should not start until #627 and #725 have established how fine a gap is real.

**Shape.** A line beneath or beside the analysis table naming what the chosen discard gave up, optionally with a compact delta inside the highlighted row's Net cell. **Not a fifth column** — `skills/ui-layout-and-interaction/SKILL.md` records that a phone-width portrait viewport cannot fit enlarged scores alongside six mini-cards, the arrow, and four numeric columns, so a fifth column would need the #637 redesign first. This shape needs neither #637 nor a column-fit fight, and works on the tightest phone, which is where the milestone's tie-breaker says it has to work.

**Size.** Small. `getDiscardQuality(...)` already computes the chosen discard's `expectedPointsLoss` on `src/ui-react/ScoredPossibleKeepDiscards.tsx` line 147, inside the very component that draws the table; today it is handed up through `onAnalysisRendered` to telemetry and the tally and never rendered. No engine, analysis, or artifact change.

## Acceptance criteria

- [ ] With a complete discard selected, the analysis view states what the chosen discard gave up against the best-scoring option, in expected points.
- [ ] An optimal choice reads as no loss, never as `+0.00` or `-0.00`, per the round-before-signing rule in `AGENTS.md`.
- [ ] The figure is the same value `getDiscardQuality` already produces, so the per-hand number and the "Lost per discard" average can never disagree.
- [ ] Nothing is shown before a discard is complete — asserted, not merely implied by when the table mounts.
- [ ] Legible in both responsive modes at an enlarged root font, with no new table column and no portrait overflow.
- [ ] No per-row delta for the fourteen unchosen options; that form belongs to #11.
- [ ] Storybook stories for the optimal and sub-optimal cases; Playwright baselines regenerated per `skills/testing-e2e/SKILL.md`.

The `needs design` label comes off with this edit: the open question is now answered.

**Comment by markafitzgerald1** [comment]
I'm not sure if we want to communicate how the user could discard better in this manner. Removing from the MLP milestone in the interim until a decision on how to proceed with this Issue is made.

**Comment by markafitzgerald1** [comment]
Claude Code (Sonnet 5, high effort), on Mark's behalf. **Design proposal — needs your call before implementation.** Posted while picking the next Todo item; #10 sits at the top of Todo but is not implementation-ready (one-line body from 2023, `needs design` label, and the unresolved product question in your 2026-05-04 comment).

## The open question is still open

Your 2026-05-04 note — *"I'm not sure if we want to communicate how the user could discard better in this manner"* — was never resolved. The 2026-09-06 milestone ranking pass put #10 back in MLP, which reads as "yes, we want it", but the *how* is undecided, and that's the whole risk: an explicit per-row "this choice costs you N points" can tip the table from *training aid* toward *answer key*, shortcutting the thinking the app is supposed to build.

**Recommendation: ship it, but only after a discard is committed.** Rationale:

- The analysis table already shows `Net` per candidate, sorted descending — the point-loss information is *already on screen*, just as a subtraction the user has to do. #10 only makes the arithmetic explicit.
- Gating the delta on "a complete discard is selected" means it never pre-empts the decision. It appears in the same moment #19's running EV-loss tally already fires and #717/#747 would show a played-out result — the *review* moment, not the *decide* moment. That keeps it on the "diagnose mistakes / understand them" side of the loop rather than the "make authentic decisions" side.
- It composes with #765 (classify *why*): #10 says *how much* this row costs, #765 says *which component* (hand/crib/play) the cost came from.

## Three concrete shapes (pick one)

1. **"vs best" delta column, post-commit only.** A 5th numeric column, U+2212-signed like the other signed columns, `0.00` on the optimal row, populated only once a complete discard is selected (blank/absent before). *Cost:* the portrait table already can't fit enlarged scores next to 6 mini-cards + arrow + 4 numeric columns (now documented in `skills/ui-layout-and-interaction/SKILL.md`), so a 5th column needs the horizontal-mini-card redesign first, or portrait drops the column and shows it only in the expanded row.
2. **Augment the `Net` cell, post-commit.** After commit, each `Net` cell shows its value plus a small `(−N.NN)` delta-from-best beneath/beside it. No new column, so no portrait-width regression. The optimal row shows `(best)` or nothing.
3. **Expanded-row only.** The per-row ▸ disclosure (which already exists) gains a "gave up N.NN vs the best discard here" line. Zero table-width impact, works on the tightest phone, but it's one tap removed from glanceable.

My lean: **2 for the collapsed table + 3's line in the expanded row**, both post-commit. Avoids the portrait-width problem entirely and still puts the number one glance away.

## #11 (colour scale)

#11 is the same feature in approximate visual form and explicitly depends on #10 for the text equivalent (accessibility). Recommend **keeping them separate but sequencing #11 immediately after #10**, sharing one `pointsLostVsBest(row, rows)` helper. Merging them into one PR risks the colour-blindness work (the substantive part of #11) getting rushed.

## Size

Small once the shape is fixed: one derived value (`rows[0].expectedNetPoints − row.expectedNetPoints` when sorted by Net; recompute the max when sorted by another key), one render change in `ScoredPossibleKeepDiscards.tsx`, a Storybook story for the pre/post-commit states, Jest assertions, and a Playwright screenshot regen. No engine, analysis, or artifact changes — `expectedNetPoints` is already computed per candidate in `src/analysis/analysis.ts`.

## What I need from you

1. Ship it, or not? (I recommend yes.)
2. Pre-commit or post-commit visibility? (I recommend post-commit.)
3. Which shape — 1, 2, 3, or the 2+3 combination I lean toward?

Once you answer, I'll fold the decision into the issue body as a dated scope edit and pick it up.

**Comment by markafitzgerald1** [comment]
Claude Code (Opus 5, high effort), on Mark's behalf: **delivered by PR #795, close when that merges.**

The 2026-09-09 scope decision above rescoped this issue to showing what the *chosen* discard gave up, rather than a delta on all fifteen rows. PR #795, which implements #765, does exactly that: the chosen row in the analysis table now carries an `Optimal discard` badge, or a `Sub-optimal: N pts lost` badge naming the dominant loss component.

That was scope expansion on #795 — it was not in #765's recorded scope and the pull request did not flag it — but Mark has accepted it, and #765's body now records the decision and the reasoning. Surfacing the diagnosis where the decision is made, rather than only later in the mistake queue, is a better fit for the loop this milestone is built around.

Leaving this open until #795 merges rather than closing now, per the convention that only a merged change carries the closing decision. If #795 is withdrawn or reverted, this issue is still live and its acceptance criteria still stand.

What remains unbuilt here is the part deliberately cut on 2026-09-09: a per-row delta for the other fourteen candidates. That belongs to #11, whose colour scale needs a text equivalent, and should not start before #627 and #725 establish how fine a gap is real.

**Comment by markafitzgerald1** [comment]
Claude Code (Opus 5, High effort), closing on Mark's behalf: delivered by #765 via PR #795, merged 2026-09-17.

This issue's 2026-09-09 scope decision asked for "a line beneath or beside the analysis table naming what the chosen discard gave up", explicitly not a delta on all fifteen rows. That is what shipped: a 'Sub-optimal: N pts lost' badge with a component breakdown beside it, for example '0.59 Hand gain < 0.63 Play + 0.25 Crib loss', where the components sum to the headline so the figure can be checked without leaving the caption.

No closing keyword was ever carried in a PR body for this, per the rule that one issue spanning several PRs waits for the final one, so it is closed by hand here.

The fifteen-row form remains out of scope and lives on #11, which should not start until #627 and #725 establish how fine a gap is real.
